frontend/packages/dev-console/src/components/monitoring/overview/MonitoringOverview.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionToggle,
+  AccordionContent,
+  Split,
+  SplitItem,
+  Badge,
+} from '@patternfly/react-core';
+import { K8sResourceKind, EventKind } from '@console/internal/module/k8s';
+import { requirePrometheus } from '@console/internal/components/graphs';
+import { workloadMetricQueries } from './queries';
+import MonitoringDashboardGraph from '../dashboard/MonitoringDashboardGraph';
+import MonitoringEventsWarning from './MonitoringEventsWarning';
+import MonitoringEvents from './MonitoringEvents';
+import './MonitoringOverview.scss';
+
+const WorkloadGraphs = requirePrometheus(({ resource }) => {
+  const ns = resource.metadata.namespace;
+  const workloadName = resource.metadata.name;
+  const workloadType = resource.kind.toLowerCase();
+  return (
+    <>
+      {_.map(workloadMetricQueries, (q) => (
+        <MonitoringDashboardGraph
+          title={q.title}
+          namespace={ns}
+          graphType={q.chartType}
+          query={q.query({ ns, workloadName, workloadType })}
+          humanize={q.humanize}
+          byteDataType={q.byteDataType}
+        />
+      ))}
+    </>
+  );
+});
+
+type MonitoringOverviewProps = {
+  resource: K8sResourceKind;
+  events: EventKind[];
+};
+
+const MonitoringOverview: React.FC<MonitoringOverviewProps> = ({ resource, events }) => {
+  const [expanded, setExpanded] = React.useState(['metrics']);
+  const eventWarning = _.filter(events, ['type', 'Warning']);
+
+  const onToggle = (id: string) => {
+    const index = expanded.indexOf(id);
+    const newExpanded =
+      index >= 0
+        ? [...expanded.slice(0, index), ...expanded.slice(index + 1, expanded.length)]
+        : [...expanded, id];
+    setExpanded(newExpanded);
+  };
+
+  return (
+    <div className="odc-monitoring-overview">
+      <Accordion
+        asDefinitionList={false}
+        noBoxShadow
+        className="odc-monitoring-overview__metric-accordion"
+        headingLevel="h5"
+      >
+        <AccordionItem>
+          <AccordionToggle
+            onClick={() => {
+              onToggle('events-warning');
+            }}
+            isExpanded={expanded.includes('events-warning')}
+            id="events-warning"
+            className="odc-monitoring-overview__event-warning-toggle"
+          >
+            <Split>
+              <SplitItem>Events (Warning)</SplitItem>
+              <SplitItem isFilled />
+              <SplitItem>
+                <Badge key={1}>{eventWarning.length}</Badge>
+              </SplitItem>
+            </Split>
+          </AccordionToggle>
+          <AccordionContent
+            className="odc-monitoring-overview__event-warning-body"
+            id="events-warning"
+            isHidden={!expanded.includes('events-warning')}
+          >
+            <MonitoringEventsWarning events={eventWarning} />
+          </AccordionContent>
+        </AccordionItem>
```

Comment:
I already added separate component for ```events (Warning)```, ```Metrics```, ```Events```. I added ```AccordianItem``` here so that state can be maintained for the accordion.